### PR TITLE
sh2.cpp : Updates for SH-1 Family product lines

### PR DESCRIPTION
**Pull request by cam900**
SH7020 lines are actually SH-1 Family at official roadmap
Internalize on-chip RAM, Add / Fix notes
Correct shadowings for SH-1 Family product lines
Add / fix configs for all variants for SH7020/SH7030 product lines
Fix namings, Spacings
References are official documents

### CODE REVIEW

**Comment by angelosa** [comment]
This is wrong and untested, so gonna close this.

I'll eventually take care of it.

**Comment by rb6502** [comment]
Do you have proof of it being wrong and untested?  I'll grant the clash part.

**Comment by angelosa** [comment]
Mirroring for sure is wrong, in the first place it must be in the accessors instead (namely for future cache implementation) plus how that mirroring is implemented is probably hit startup performance for no reason. Plus how the internal i/o shouldn't be split into different CPUs but must be delegated into serviceable callbacks, sort of bus devices like they really are.

SH1 / 2A is really currently just used by Cool Riders, so I'm not going to eventually fix breakups for more juicy stuff like Saturn CDB and Casio Loopy.

Bottom line I dislike how he keeps doing stuff on which he clearly looks at from the commits and having no clue or interest about how to kill tech debt, it's not how it works.
